File: tools/calculate_fft_structure.m
% return the index and W of m*n FFT structure.
function [index w]=calculate_fft_structure(m,n)
r=0:m*n-1;
index=reshape(r,[m,n]);
w=zeros(m,n);
for row=1:m
    for col=1:n
        w(row,col)=exp(-1i*2*pi*(row-1)*(col-1)/(m*n));
    end
end
w=w(:);

File: code_generator/calculate_fft_structure.py
import numpy as np
import cmath


def calculate_fft_structure(m, n):
    index = np.arange(m * n).reshape(m, n)
    w = np.ndarray(shape=(m, n), dtype=complex)
    for row in xrange(m):
        for col in xrange(n):
            w[row][
                col] = cmath.exp(-2j * cmath.pi * row * col / complex(m * n))
    return index, w

if __name__ == '__main__':
    index, w = calculate_fft_structure(4, 2)
    for i in xrange(8):
        print w[i/2,i%2]
    print w
    print index
